synapse/storage/types.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
```

Comment:
Aren't the classes described in this file supposed to be abstract classes?

Reply:
well, they are `Protocol`s. There's a whole bunch of stuff about that at https://mypy.readthedocs.io/en/stable/protocols.html, but basically it's a way of defining an interface that is implemented by other classes, rather than anything that is meant to be used as a base class for anything else.

Reply:
Oooh TIL.